basher.go

```diff
@@ -232,10 +233,22 @@ func (c *Context) Run(command string, args []string) (int, error) {
 	for _, arg := range args {
 		argstring = argstring + " '" + arg + "'"
 	}
+
+	signals := make(chan os.Signal, 1)
+	signal.Notify(signals)
+
 	cmd := exec.Command(c.BashPath, "-c", command+argstring)
 	cmd.Env = []string{"BASH_ENV=" + envfile}
 	cmd.Stdin = c.Stdin
 	cmd.Stdout = c.Stdout
 	cmd.Stderr = c.Stderr
-	return exitStatus(cmd.Run())
+	cmd.Start()
+	go func() {
+		for sig := range signals {
+			if !cmd.ProcessState.Exited() {
```

Comment:
@progrium this check throws the following error. i'm, obviously, doing something wrong here...

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x8 pc=0x45d3e5]

goroutine 6 [running]:
os.(*ProcessState).exited(0x0, 0x101)
	/usr/src/go/src/os/exec_posix.go:69 +0x5
os.(*ProcessState).Exited(0x0, 0xc20806a120)
	/usr/src/go/src/os/doc.go:66 +0x28
github.com/michaelshobbs/go-basher.func·001()
	/go/src/github.com/michaelshobbs/go-basher/basher.go:248 +0x9d
created by github.com/michaelshobbs/go-basher.(*Context).Run
	/go/src/github.com/michaelshobbs/go-basher/basher.go:252 +0x635

goroutine 1 [syscall]:
syscall.Syscall6(0x3d, 0x9, 0xc208093a2c, 0x0, 0xc20806c090, 0x0, 0x0, 0x7f85eb8ee000, 0x407051, 0xc20806a1a0)
	/usr/src/go/src/syscall/asm_linux_amd64.s:46 +0x5
syscall.wait4(0x9, 0xc208093a2c, 0x0, 0xc20806c090, 0x90, 0x0, 0x0)
	/usr/src/go/src/syscall/zsyscall_linux_amd64.go:124 +0x79
syscall.Wait4(0x9, 0xc208093a74, 0x0, 0xc20806c090, 0x0, 0x0, 0x0)
	/usr/src/go/src/syscall/syscall_linux.go:224 +0x60
os.(*Process).wait(0xc20800a3c0, 0x0, 0x0, 0x0)
	/usr/src/go/src/os/exec_unix.go:22 +0x103
os.(*Process).Wait(0xc20800a3c0, 0xc20803c4e0, 0x0, 0x0)
	/usr/src/go/src/os/doc.go:45 +0x3a
os/exec.(*Cmd).Wait(0xc20807a000, 0x0, 0x0)
	/usr/src/go/src/os/exec/exec.go:364 +0x23c
github.com/michaelshobbs/go-basher.(*Context).Run(0xc20803a140, 0x593890, 0x4, 0xc20800a010, 0x1, 0x1, 0x0, 0x0, 0x0)
	/go/src/github.com/michaelshobbs/go-basher/basher.go:253 +0x648
github.com/michaelshobbs/go-basher.Application(0xc20803c840, 0xc208093f30, 0x6, 0x6, 0x5d7b18, 0x1)
	/go/src/github.com/michaelshobbs/go-basher/basher.go:74 +0x753
main.main()
	/usr/src/myapp/herokuish.go:93 +0x1d3

goroutine 5 [syscall]:
os/signal.loop()
	/usr/src/go/src/os/signal/signal_unix.go:21 +0x1f
created by os/signal.init·1
	/usr/src/go/src/os/signal/signal_unix.go:27 +0x35
```